### DISCUSSION TITLE
reorderMesh optimal unique vertices for normal always facing outward.

### DISCUSSION THREAD

**Discussion by RyugaRyuzaki** [zeux/meshoptimizer#784] [General]
In threejs I have this : 
```
const position = new Float32Array([....])
const normal= new Float32Array([....])
const index= new Uint32Array([....])
```
Those were not compressed or reordered.
Now i want to use
``` 
reorderMesh: (indices: Uint32Array, triangles: boolean, optsize: boolean) => [Uint32Array, number];
```
With output is the normal vector will be always  facing outward likes these images .

![Capture1](https://github.com/user-attachments/assets/0dd18470-675e-46e9-9f8c-c2cfc6ac3fe5)

![Capture2](https://github.com/user-attachments/assets/bec6bcfb-25a8-485d-aa70-8144d519e0c8)

How can i do it? Many thank!

**Comment by zeux** [comment]
Sorry, I'm not sure what you mean. reorderMesh has no relevance on the normal values, it only changes the order of vertices/triangles, and does so without changing the winding of triangles. If the screenshot you get is after reorderMesh, and it looks properly before it, then you have a bug in the code that calls it.
